modules/smil-workflowoperation/src/main/java/org/opencastproject/workflow/handler/smil/CutMarksToSmilWorkflowOperationHandler.java

```diff
@@ -0,0 +1,303 @@
+        logger.info("Skipped mark with begin: {}, ", begin);
+        iter.remove();
+      }
+    }
+    // If the timestamp list is now empty, give up
+    if (cutMarks.size() < 1) {
+      logger.warn("No timestamps are valid for the given tracks! Skipping...");
+      return createResult(mediaPackage, WorkflowOperationResult.Action.SKIP);
+    }
+
+    // Create the new SMIL document
+    Smil smil;
+    try {
+      SmilResponse smilResponse = smilService.createNewSmil(mediaPackage);
+
+      logger.info("Start adding tracks");
+      for (Times mark : cutMarks) {
+        smilResponse = smilService.addParallel(smilResponse.getSmil());
+        SmilMediaContainer par = (SmilMediaContainer) smilResponse.getEntity();
+        logger.debug("Segment begin: {}; Segment duration: {}", mark.begin, mark.duration);
+        // Add tracks (as array) to par
+        smilResponse = smilService
+                .addClips(smilResponse.getSmil(),
+                        par.getId(),
+                        tracksFromFlavors.toArray(new Track[0]),
+                        mark.begin,
+                        mark.duration);
+      }
+
+      smil = smilResponse.getSmil();
+      logger.info("Done adding tracks");
+    } catch (SmilException e) {
+      throw new WorkflowOperationException("Failed to create SMIL Catalog", e);
+    }
+
+    // Put new SMIL into workspace and add it to media package
+    try (InputStream is = IOUtils.toInputStream(smil.toXML(), "UTF-8")) {
+      URI smilURI = workspace.put(mediaPackage.getIdentifier().toString(), smil.getId(), CUTTING_SMIL_NAME, is);
+      MediaPackageElementBuilder mpeBuilder = MediaPackageElementBuilderFactory.newInstance().newElementBuilder();
+      Catalog catalog = (Catalog) mpeBuilder
+              .elementFromURI(smilURI, MediaPackageElement.Type.Catalog, targetSmilFlavor);
+      catalog.setIdentifier(smil.getId());
+      for (String tag : targetTags) {
+        catalog.addTag(tag);
+      }
+      mediaPackage.add(catalog);
+    } catch (JAXBException | SAXException | IOException e) {
+      throw new WorkflowOperationException("Failed to parse crated SMIL Catalog", e);
+    }
+
+    final WorkflowOperationResult result = createResult(mediaPackage, WorkflowOperationResult.Action.CONTINUE);
+    logger.debug("Cut marks to smil operation completed");
+    return result;
+  }
+
+  private List<Track> getTracksFromFlavor(MediaPackageElementFlavor flavor, MediaPackage mediaPackage) {
+    // Get tracks from flavor
+    TrackSelector trackSelector = new TrackSelector();
+    trackSelector.addFlavor(flavor);
+    Collection<Track> tracks = trackSelector.select(mediaPackage, false);
+
+    // Get only videos
+    ArrayList<Track> videos = new ArrayList<>();
+    for (Track video : tracks) {
+      if (video.hasVideo()) {
+        videos.add((video));
+      } else {
+        logger.debug("Skipping track {} since it does not seem to have a vide track.", video);
+      }
+    }
```

Comment:
I'm wondering if there is a specific reason to require video streams to be present. This would mean that we cannot cut audio and also that we need to make sure that we run `inspect` on the media package before we run this operation.

Reply:
I honestly do not remember why only video tracks were allowed (should've commented on that, arg). I'll try and see if I can't figure it out.

Reply:
Whatever the reason was that I originally implemented this, it seems to have vanished.  So I suppose this code snippet could be removed.

Reply:
Removed the filter. Now it's just the track selector.